File: src/main/java/uk/co/eelpieconsulting/osm/nominatim/indexing/JsonSerializer.java
package uk.co.eelpieconsulting.osm.nominatim.indexing;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.SerializationFeature;

class JsonSerializer {

	private ObjectMapper mapper;

	public JsonSerializer() {
		mapper = new ObjectMapper();
        mapper.configure(SerializationFeature.FAIL_ON_EMPTY_BEANS, false);
        mapper.configure(SerializationFeature.WRITE_NULL_MAP_VALUES, false);
	}
	
	public String serialize(Object object) {
		try {
			return mapper.writeValueAsString(object);
		} catch (JsonProcessingException e) {
			throw new RuntimeException(e);
		}
	}
	
}

File: src/main/kotlin/uk/co/eelpieconsulting/osm/nominatim/json/JsonSerializer.kt
package uk.co.eelpieconsulting.osm.nominatim.json

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.module.kotlin.registerKotlinModule
import org.springframework.stereotype.Component
import uk.co.eelpieconsulting.osm.nominatim.model.Place

@Component
class JsonSerializer {

    private val mapper: ObjectMapper = ObjectMapper().registerKotlinModule()

    fun serializePlace(place: Place): String {
        return mapper.writeValueAsString(place)
    }

}
